cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -142,7 +147,7 @@ protected int getInstances(HttpRequest request) throws IllegalArgumentException,
     Reader reader = new InputStreamReader(new ChannelBufferInputStream(content), Charsets.UTF_8);
     try {
       return GSON.fromJson(reader, type);
-    } catch (JsonSyntaxException e) {
+    } catch (Exception e) {
```

Comment:
`GSON.fromJson()` can fail with a `RuntimeException` due to a missing no-args constructor, and it's useful to see the stack trace from the test run. Can change to `RuntimeException`.